bin/pre-commit

```diff
@@ -1,33 +1,53 @@
 #!/bin/sh
 
-PATH="$PATH:/usr/local/bin"
-
 printf "\nBy contributing to this project, you license the materials you contribute under the GNU General Public License v2 (or later). All materials must have GPLv2 compatible licenses — see .github/CONTRIBUTING.md for details.\n\n"
 
-files=$(git diff --cached --name-only --diff-filter=ACM | grep ".jsx*$")
-if [ "$files" = "" ]; then
-    exit 0
-fi
-
-pass=true
-
-printf "\nValidating .jsx and .js:\n"
-
-for file in ${files}; do
-    ./node_modules/.bin/eslint --cache ${file}
-    if [ $? -ne 0 ]; then
-        printf "\033[31meslint Failed: %s\033[0m\n" "${file}"
-        pass=false
-    else
-        printf "\033[32meslint Passed: %s\033[0m\n" "${file}"
-    fi
-done
-
-printf "\neslint validation complete\n"
-
-if ! $pass; then
-    printf "\n\033[41mCOMMIT FAILED:\033[0m Your commit contains files that should pass validation tests but do not. Please fix the errors and try again.\n\n"
-    exit 1
+# We want to know whether this script was called from the CLI or from a GUI app.
+# If it is the later, we would only show the ESLint command to get the report,
+# not the report itself, because GUIs tend to left a small space for pre-commit
+# messages and they are not useful. At least this way, the user can see something.
+#
+# - First, get the caller: the process that executes the commit.
+#	The command tree would be: caller > git > pre-commit script
+#       so we need to get the grandparent of this script
+#
+# - Second, if the caller is not one of the available shells (see /etc/shells),
+#	we are going to assume that it is called from a GUI app.
+
+# caller=$(which $(ps -o comm=  $(ps -o ppid= $(ps -o ppid= $$))))
+caller=$(which $(ps -o comm=  $(ps -o ppid= $PPID)))
```

Comment:
Could checking whether the grand parent process has a tty attached be more reliable?